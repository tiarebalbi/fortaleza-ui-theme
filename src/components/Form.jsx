// @flow

import * as React from 'react';
import { Form as AntDForm } from 'antd';
import type { FormType } from '../types';

const Form = ({ layout = "vertical" }: FormType) => (<AntDForm layout={ layout } />);

export default Form;
