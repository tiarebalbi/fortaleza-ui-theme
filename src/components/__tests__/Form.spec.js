import * as React from 'react';
import { shallow } from 'enzyme';

import Form from '../Form';

describe('Form', () => {
  const wrapper = shallow(<Form />);

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should build component using default properties', () => {
    expect(wrapper.props().layout).toBe('vertical');
  });

  it('should build component with custom layout', () => {
    const localWrapper = shallow(<Form layout="inline" />);
    expect(localWrapper.props().layout).toBe('inline');
  });
});
