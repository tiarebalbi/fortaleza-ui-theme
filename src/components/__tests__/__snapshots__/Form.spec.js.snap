// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form should match snapshot 1`] = `
<Form
  hideRequiredMark={false}
  layout="vertical"
  onSubmit={[Function]}
  prefixCls="ant-form"
/>
`;
